src/main/java/net/neoforged/neoforge/event/village/WandererTradesEvent.java

```diff
@@ -21,10 +22,12 @@
 public class WandererTradesEvent extends Event {
     protected List<ItemListing> generic;
     protected List<ItemListing> rare;
+    private final RegistryAccess registryAccess;
 
-    public WandererTradesEvent(List<ItemListing> generic, List<ItemListing> rare) {
+    public WandererTradesEvent(List<ItemListing> generic, List<ItemListing> rare, RegistryAccess registryAccess) {
```

Comment:
The previous constructor needs to be retained and marked as deprecated with a comment since it's not marked as internal and therefore has to be considered API in terms of breaking changes. It should delegate to the new constructor with `RegistryAccess.EMPTY`.